p2p/host/basic/basic_host.go

```diff
@@ -174,6 +175,8 @@ func (h *BasicHost) NewStream(pid protocol.ID, p peer.ID) (inet.Stream, error) {
 // h.Network.Dial, and block until a connection is open, or an error is
 // returned. // TODO: Relay + NAT.
 func (h *BasicHost) Connect(ctx context.Context, pi peer.PeerInfo) error {
+	ctx, cancel := context.WithTimeout(ctx, host.DiscoveryConnTimeout)
+	defer cancel()
```

Comment:
hm, i think @whyrusleeping's right. 

the timeout was originally for "discovered" peers, right? maybe it makes sense for it to be there.